src/Symfony/Component/HttpClient/Response/NativeResponse.php

```diff
@@ -130,6 +132,10 @@ private function open(): void
                 // Send request and follow redirects when needed
                 $this->handle = $h = fopen($url, 'r', false, $this->context);
                 self::addResponseHeaders($http_response_header, $this->info, $this->headers, $this->info['debug']);
+                if (null !== $this->bufferRegex && isset($this->headers['content-type'])) {
```

Comment:
same here: this should be outside the redirection loop

Reply:
Moved it outside the loop